vector/src/main/java/im/vector/app/features/location/LocationTracker.kt

```diff
@@ -88,7 +89,24 @@ class LocationTracker @Inject constructor(
     fun stop() {
         Timber.d("## LocationTracker. stop()")
         locationManager?.removeUpdates(this)
-        callback = null
+        callbacks.clear()
+    }
+
+    fun addCallback(callback: Callback) {
+        synchronized(callbacks) {
```

Comment:
if we only call `addCallback/removeCallback` from the main thread, we could avoid needing to use `synchronized`

if we are calling from multiple threads, we'll probably want to make all the callback interactions synchronised :cry:

Reply:
You are right. It is called in a single UI thread.